pallets/alt-vr/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+            username: ternoa::String,
+        ) -> DispatchResultWithPostInfo {
+            let owner = ensure_signed(origin)?;
+
+            let username_lower_bound = username.len() >= T::MinNameLength::get() as usize;
+            let username_upper_bound = username.len() <= T::MaxNameLength::get() as usize;
+
+            ensure!(username_lower_bound, Error::<T>::TooShortUsername);
+            ensure!(username_upper_bound, Error::<T>::TooLongUsername);
+
+            Altvrs::<T>::try_mutate(owner.clone(), |res| -> Result<(), Error<T>> {
+                let altvrdata = res.as_mut().ok_or(Error::<T>::UserNotFound)?;
+                altvrdata.username = username.clone();
+                Ok(())
+            })?;
+
+            let event = Event::AltvrUsernameUpdated(owner, username);
+            Self::deposit_event(event);
+
+            Ok(().into())
+        }
+
+        /// Update Altvr vchatname of the caller
+        #[pallet::weight(T::WeightInfo::update_vchatname())]
+        #[transactional]
+        pub fn update_vchatname(
+            origin: OriginFor<T>,
+            vchatname: ternoa::String,
+        ) -> DispatchResultWithPostInfo {
+            let owner = ensure_signed(origin)?;
+
+            let username_lower_bound = vchatname.len() >= T::MinNameLength::get() as usize;
+            let username_upper_bound = vchatname.len() <= T::MaxNameLength::get() as usize;
+
+            ensure!(username_lower_bound, Error::<T>::TooShortUsername);
+            ensure!(username_upper_bound, Error::<T>::TooLongVchatname);
+
+            Altvrs::<T>::try_mutate(owner.clone(), |res| -> Result<(), Error<T>> {
+                let altvrdata = res.as_mut().ok_or(Error::<T>::UserNotFound)?;
+                altvrdata.vchatname = vchatname.clone();
+                Ok(())
+            })?;
+
+            let event = Event::AltvrVchatnameUpdated(owner, vchatname);
+            Self::deposit_event(event);
+
+            Ok(().into())
+        }
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Altvr was created. \[owner, username, vchatname\]
+        AltvrCreated(T::AccountId, ternoa::String, ternoa::String),
+        /// Altvr username updated  \[owner, username\]
+        AltvrUsernameUpdated(T::AccountId, ternoa::String),
+        /// Altvr vchatname updated  \[owner, vchatname\]
+        AltvrVchatnameUpdated(T::AccountId, ternoa::String),
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Altvr username is too short.
+        TooShortUsername,
+        /// Altvr username is too long.
+        TooLongUsername,
+        /// Vchat name is too short.
+        TooShortVchatname,
+        /// Vchat name is too long.
+        TooLongVchatname,
+        /// User not found
+        UserNotFound,
+    }
+
+    /// List of Altvr datas create.
+    #[pallet::storage]
+    #[pallet::getter(fn altvrs)]
+    pub type Altvrs<T: Config> =
```

Comment:
Maybe `Users` would make for sense? AltVR is the platform and we need to track users, hence `Users`.

Reply:
done